udata/core/dataset/api.py

```diff
@@ -132,6 +134,15 @@ def parse_filters(datasets, args):
             datasets = datasets.filter(resources__schema__name=args['schema'])
         if args.get('schema_version'):
             datasets = datasets.filter(resources__schema__version=args['schema_version'])
+        if args.get('topic'):
+            if not ObjectId.is_valid(args['topic']):
+                api.abort(400, 'Topic arg must be an identifier')
+            try:
+                topic = Topic.objects.get(id=args['topic'])
+            except Topic.DoesNotExist:
+                pass
+            else:
+                datasets = datasets.filter(id__in=[d.id for d in topic.datasets])
```

Comment:
There may be performance issue here. Not sure if adding datasets ids in Topics index could help?

Reply:
```diff
diff --git a/udata/core/topic/models.py b/udata/core/topic/models.py
index c43dec0e8..f076f1134 100644
--- a/udata/core/topic/models.py
+++ b/udata/core/topic/models.py
@@ -34,7 +34,9 @@ class Topic(db.Document, db.Owned):
         'indexes': [
             '$name',
             'created_at',
-            'slug'
+            'slug',
+            'datasets.id',
+            'reuses.id',
         ] + db.Owned.meta['indexes'],
         'ordering': ['-created_at'],
         'auto_create_index_on_save': True
```

Adding this to the model does not change anything on my local env for a Topic with 5000 Datasets attached. FYI the overhead of the topic filter is ~1 second on a total of 6 seconds (yeah local env is slow).

Not actually sure how/when indexes are supposed to be computed on this ~~mock of a~~ DB, tried this to be sure:

```
>>> Topic.compare_indexes()
{'missing': [], 'extra': [[('datasets._id', 1)], [('reuses._id', 1)]]}
>>> Topic.ensure_indexes()
>>> Topic.compare_indexes()
{'missing': [], 'extra': [[('datasets._id', 1)], [('reuses._id', 1)]]}
```

I want to believe that `id` based filtering is supposed to come pre-optimized but I actually have no idea.

Reply:
Why is your local env this slow?
Tested locally with a topic with 15K attached, the entire request takes up to 300ms. Let's skip the index idea!

Reply:
I don't know 😅 Did not have time to dig.

What's your value of `FLASK_CACHE`?

I think I recall from 6 years ago that I had to dig through a confusing network configuration on my local env to speed things up, but I don't remember what it was. FYI I was a brand new env and machine since the last time I worked on udata.

Reply:
Not sure but I think @quaxsze had some performance issues at some point as well.
I have `CACHE_TYPE` set to `flask_caching.backends.null`.

Reply:
For reference (not the best place for that I guess :p) I found the "network issue" I was talking about on local (I'm using `dev.local`): https://superuser.com/a/1596341

Reply:
Never written so much doc for this project (should have done it 6 years ago!): https://github.com/opendatateam/udata/pull/2923